common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -666,18 +668,13 @@ private boolean await0(long timeoutNanos, boolean interruptable) throws Interrup
 
         checkDeadLock();
 
-        long startTime = System.nanoTime();
-        long waitTime = timeoutNanos;
-        boolean interrupted = false;
-        try {
-            for (;;) {
-                synchronized (this) {
-                    if (isDone()) {
-                        return true;
-                    }
+        synchronized (this) {
+            boolean interrupted = false;
+            try {
+                while (!isDone() && (timeoutNanos -= System.nanoTime() - startTime) > 0) {
```

Comment:
`timeoutNanos -= System.nanoTime() - startTime`

This does not look right as it is double decrementing on each iteration. Previously code was not writing back to `timeoutNanos`

Reply:
You are right, I corrected it.